### PR TITLE
Refactor and final clean up

### PR DESCRIPTION
**Pull request by stephaniemagdic**
# Thank you for pushing your code!

## What (if any) features are you implementing?
 + NA

## What (if anything) did you refactor?
  + This is the final refactoring of the javascript aside from the hydration functions.
  + ESLint was run and spacing was fixed.
  + Repetitive code statements were removed.
  + Unused files were deleted.
  + The load page function and fetch functions were changed from using async await to just using promises and .then() in order to have a good example and solid footing for moving into the next project before Mod 3. See code below: 
 
function loadPage() {
  Promise.resolve(fetchPageData()).then((data) => generateRepoClasses(data))
    .then(() => displayPageInfo());
}

`function fetchPageData() {
  const userRepoPromise = fetchData('users') 
  const hydrationRepoPromise = fetchData('hydration')  
  const sleepRepoPromise = fetchData('sleep')  
  return Promise.all([userRepoPromise, hydrationRepoPromise, sleepRepoPromise]).then(values => values);
}

function generateRepoClasses(dataSets) {
  allUserData = new UserRepository(dataSets[0].userData);
  allHydrationData = new HydrationRepository(dataSets[1].hydrationData);
  allSleepData = new SleepRepository(dataSets[2].sleepData);
}

function fetchData(type) {
  return fetch(`http://localhost:3001/api/v1/${type}`)
    .then(response => response.json())
    .then(data => data)
    .catch(err => console.log(`ERROR with ${type}: ${err}`))
}
`

- `Functions are no longer asynchronous`
- `There for the await operator is no longer used`
- `Promises are instead chained`
- `A promise.resolve was used to start a promise chain`

## Were there any issues that arose?
 + NA

## Is there anything that you need from your teammate?
 + Merge this PR together so we can look at the conflicts that might arise from the hydration functionality refactoring.

## Any other comments, questions, or concerns?
 + NA

### CODE REVIEW

File: src/HydrationRepository.js [hunk 1]
```diff
@@ -3,9 +3,9 @@ class HydrationRepository {
     this.hydroData = hydroData;
   }
 
- returnUserData(id) {
-   return this.hydroData.filter(hydroItem => hydroItem.userID === id)
- }
+  returnUserData(id) {
```
Comment (alyssajoyner) [inline src/HydrationRepository.js:6]: The name of this function is odd to me and doesn't really describe what it is doing. I would rename it to something like `findHydroItem`.

File: src/UserRepository.js [hunk 2]
```diff
@@ -13,7 +13,7 @@ class UserRepository {
       return totalSteps 
     }, 0)
     return average / this.userData.length
```
Comment (alyssajoyner) [inline src/UserRepository.js:15]: Why is this method in the UserRepository? Should it not be in Activity.js?

File: src/scripts.js [hunk 3]
```diff
@@ -25,57 +25,40 @@ const dailyStepGoal = document.querySelector(".daily-step-goal-num")
...
-      //  .catch(err => console.log(console.log(errorMessage));
-
   const sleepRepoPromise = fetchData('sleep')  
-    // .then(console.log('success'));
-      //  .catch(err => /* do something else */);
-
-  const apiDataSets = await Promise.all([userRepoPromise, hydrationRepoPromise, sleepRepoPromise]).then(values => values);
-
-  return apiDataSets;
+  return Promise.all([userRepoPromise, hydrationRepoPromise, sleepRepoPromise]).then(values => values);
```
Comment (alyssajoyner) [inline src/scripts.js:44]: Great use of Promise.all!

File: src/scripts.js [hunk 4]
```diff
@@ -25,57 +25,40 @@ const dailyStepGoal = document.querySelector(".daily-step-goal-num")
...
 }
 
 function generateRepoClasses(dataSets) {
-  //TO DO: Refactor-loop through argument to dry up.
```
Comment (alyssajoyner) [inline src/scripts.js:63]: Love this To Do - great opportunity to dry up code and make it more dynamic.

File: test/UserRepository-test.js [hunk 5]
```diff
@@ -14,7 +14,7 @@ describe('User Repository', () => {
       16,
       4,
       8
-      ]}, {
+    ]}, {
     id: 2,
     name: 'Jarvis Considine',
```
Comment (alyssajoyner) [inline test/UserRepository-test.js:19]: Ensure that the info you use in tests is made up and fake (and different from the info provided). This is done for privacy/security purposes.

**Review by alyssajoyner — COMMENTED** [review]
Nice work! Please let me know if you have any questions or want to hop on a call to discuss.

**Review by sarahrudy — APPROVED** [review]
Reviewed, refactored, and pushed final changes as a team.

**Comment by sarahrudy** [comment]
Mentor feedback is super helpful. Let's revisit this and incorporate into future projects!

**Pull request merged**